runtime/common/src/fees.rs

```diff
@@ -116,63 +128,117 @@ macro_rules! impl_foreign_fees {
             Ok(converted_fee)
         }
 
+        fn get_fee_factor_campaign_asset(
+            campaign_asset: CampaignAsset,
+        ) -> Result<Balance, TransactionValidityError> {
+            let ztg_supply = Balances::total_issuance();
+            let campaign_asset_supply = AssetManager::total_issuance(campaign_asset.into());
+            let fee_multiplier = Balance::from(CampaignAssetFeeMultiplier::get());
+
+            // Use neutral fee multiplier if the ZTG supply is 100x greater than the campaign
+            // asset supply.
+            if ztg_supply.saturating_div(campaign_asset_supply) >= fee_multiplier {
+                Ok(BASE)
+            } else {
+                campaign_asset_supply.saturating_mul(fee_multiplier).bdiv(ztg_supply).map_err(
```

Comment:
Shall `FixedMulDiv` be used here? https://github.com/zeitgeistpm/zeitgeist/blob/4d3519c31b52e89dcbff2f992fdc478cc7a3f054/primitives/src/math/fixed.rs#L93-L107

Reply:
Assume the campaign_asset_supply is huge, then `saturating_mul` would lead to the biggest `u128`. Is the maximum campaign asset supply capped too?

Reply:
The maximum campaign asset supply is not capped. I don't see any obvious issues with a supply of u128::MAX, as math seems to work anyways, fees are just reduced by up to `CampaignAssetFeeMultiplier`. If malicious behavior is detected council and technical committee can nuke the asset. Generally, it is probably advisable to assign admin accounts to either Zeitgeist or trusted accounts.
In regards to `FixedMulDiv`, I think `bmul_bdiv` is the wrong function here, because `CampaignAssetFeeMultiplier` has  base 10 instead of `BASE` which `bmul_bdiv` requires.

Reply:
One way to improve this, if we're interested in that, is to check for overflow when multiplying and divide first if the check is positive. Not sure if we care about that though.